housenumparser/elements.py

```diff
@@ -1,382 +1,353 @@
 # -*- coding: utf-8 -*-
-
-'''
-A housenumber element.
-This is an abstract superclass for all output of the housenumberreader.
-This can be a housenumber, a housenumber series, readingerror, etc.
-'''
-
-class Element():
-
-    '''
-    :param h1: Integer first huisnummer.
-    :param bisn1: String first bisnummer.
-    :param bisl1: String first bisletter.
-    :param busn1: String first busnummer.
-    :param busl1: String first busletter.
-    :param h2: Integer last huisnummer of the series.
-    :param bisn2: String last bisnummer of the series.
-    :param bisl2: String last bisletter of the series.
-    :param busn2: String last busnummer of the series.
-    :param busl2: String last busletter of the series.
-    '''
-    def __init__(
-        self, h1, bisn1=-1, bisl1=-1, busn1=-1, busl1=-1,
-        h2=-1, bisn2=-1, bisl2=-1, busn2=-1, busl2=-1
-    ):
-            self.data = [
-                h1, bisn1, bisl1, busn1, busl1,
-                h2, bisn2, bisl2, busn2, busl2
-            ]
-
-    def getData(self, i):
-        '''
-        :param i: Integer index of the required data
-        :returns: Integer or String
-        '''
-        return self.data[i]
-
-
-'''
-Class for a readingerror in the housenumberreader.
-'''
+import re
+from enum import Enum
+
+
+class BadInput(Enum):
+    RAISE = 1  # Raises exception on bad input
+    IGNORE = 2  # A element with str() error message info will be returned
```

Comment:
Misschien is IGNORE niet zo'n goeie naam, maar ik kan niet direct iets beter verzinnen.

Reply:
Especially with `DROP`:  _"the error will be **ignored**, and..."_

Reply:
mja, misschien dat ik er `PRINT` of `SHOW` van maak.
Hier zie je de resultaten van de soorten:
https://github.com/OnroerendErfgoed/housenum-be-r-parser/blob/5b8feec2ef5b2b4d385da7bcc6a34236a62947cf/tests/test_split.py#L234-L258

Nu ik de resultaten van IGNORE zie, denk ik dat ik beter de originele input data erbij zet.

Reply:
Mijn voorstel is ERROR_MSG of ERROR.